rewriter/Struct.cc

```diff
@@ -22,30 +22,30 @@ bool isKeywordInitKey(const core::GlobalState &gs, const ast::ExpressionPtr &nod
     return false;
 }
 
-bool isMissingInitialize(const core::GlobalState &gs, const ast::Send *send) {
-    if (!send->hasBlock()) {
-        return true;
-    }
-
-    auto block = send->block();
-
-    if (auto *insSeq = ast::cast_tree<ast::InsSeq>(block->body)) {
-        auto methodDef = ast::cast_tree<ast::MethodDef>(insSeq->expr);
-
-        if (methodDef && methodDef->name == core::Names::initialize()) {
-            return false;
-        }
+// Elem = type_member {{fixed: T.untyped}}
+ast::ExpressionPtr elemFixedUntyped(core::LocOffsets loc) {
+    auto typeMember = ast::MK::Send0(loc, ast::MK::Self(loc), core::Names::typeMember(), loc.copyWithZeroLength());
+    ast::cast_tree_nonnull<ast::Send>(typeMember)
+        .setBlock(ast::MK::Block0(
+            loc, ast::MK::Hash1(loc, ast::MK::Symbol(loc, core::Names::fixed()), ast::MK::Untyped(loc))));
+    return ast::MK::Assign(loc, ast::MK::UnresolvedConstant(loc, ast::MK::EmptyTree(), core::Names::Constants::Elem()),
+                           std::move(typeMember));
+}
 
-        for (auto &&stat : insSeq->stats) {
-            methodDef = ast::cast_tree<ast::MethodDef>(stat);
+void selfScopeToEmptyTree(ast::UnresolvedConstantLit &cnst) {
```

Comment:
This function was required because we have a test called `struct_self.rb` that tests

```ruby
self::A = Struct.new(:foo)
```

and then we had an ENFORCE that said that everything in the `ancestors` list is always a ConstantLit or UnresolvedConstantLit.